### PR TITLE
- add libconfig/1.7.2

### PR DESCRIPTION
**Pull request by SSE4**
Specify library name and version:  **libconfig/1.7.2**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

I am mainly testing my templates and scripts on some libraries.

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/libconfig/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
```
Comment (prince-chrismc) [inline recipes/libconfig/all/conanfile.py:30]: Should this be prefered over the _old_ way?

**Review by prince-chrismc — APPROVED** [review]

File: recipes/libconfig/all/conanfile.py (hunk 1 above)
Comment (SSE4) [reply recipes/libconfig/all/conanfile.py:30]: sure, why not?

Comment (prince-chrismc) [reply recipes/libconfig/all/conanfile.py:30]: awesome 👍

Comment (uilianries) [reply recipes/libconfig/all/conanfile.py:30]: Yes. We added `strip_root` for this propose ;)

**Review by uilianries — APPROVED** [review]

**Review by jgsogo — APPROVED** [review]
Your templates work really well  🙌

**Pull request merged**